src/common/components/repositories-home/index.js

```diff
@@ -53,10 +53,14 @@ class RepositoriesHome extends Component {
     // if snaps stopped fetching
     if ((this.props.snaps.isFetching !== snaps.isFetching) && !snaps.isFetching) {
       // if user doesn't have enabled repos open add repositories view
-      if (!hasSnaps) {
+      if (!hasSnaps && nextProps.hasJustSignedIn) {
         this.props.router.replace('/select-repositories');
       } else {
-      // or update builds for their snaps
+        // or remove sign-in from URL
+        if (nextProps.hasJustSignedIn) {
+          this.props.router.replace(`/user/${nextProps.user.login}`);
```

Comment:
Maybe it would be less of a time-bomb for the future to just remove `sign-in` from the query string rather than providing an entire replacement URL here?